### ISSUE TITLE
Transparency render bug

### ISSUE THREAD

**Issue by Askaniy** [CelestiaProject/Celestia#1601]
**Screenshots**
![изображение](https://user-images.githubusercontent.com/44873271/220948414-7a662f89-cd48-4426-8405-7514106e1362.png)

**Desktop**
 - OS: Linux (openSUSE Tumbleweed + KDE 5.27 + Wayland)
 - Frontend: Qt 5
 - Version: 1.7.0 b0f40cec084d85ce767e35d9f2d344fe12b72be1

**Comment by levinli303** [comment]
probably same issues as #1551

**Comment by 375gnu** [comment]
if you revert  #1594 does it help?
```
git revert 69be12027e30d60d7861f6c6954cb15e4d1c4a01
```

**Comment by Askaniy** [comment]
Can't revert it easily, I use compiled versions provided by [this repository](https://download.opensuse.org/repositories/home:/munix9:/unstable/openSUSE_Tumbleweed/).

**Comment by munix9** [comment]
I will try to test it this weekend, but my hardware is currently a bit unstable.

Have you tested the AppImage? It is basically the same code base, but compiled under openSUSE Leap 15.4 - maybe you get a different result.

**Comment by Askaniy** [comment]
I hope I did what you meant. It seems to work there.
![screenshot](https://user-images.githubusercontent.com/44873271/221292576-8066a24f-b984-4886-a0c6-731cb01a0fb2.png)

**Comment by munix9** [comment]
Yes, that's what I meant, thank you.
Now we know that in general it works on openSUSE, but on Tumbleweed apparently there is "currently" a problem.
openSUSE Tumbleweed is a rolling release, so it is quite possible that a new os component/library can cause problems.
Let's see if I get something out this weekend.

**Comment by munix9** [comment]
Ok, I can reproduce it:

openSUSE Tumbleweed 20230223
Celestia 1.7.0~git20230224+15df4ba

In a Plasma-X11 session the display is correct.
In a Plasma-Wayland session the rendering of the clouds is wrong, i.e. without transparency.

This is both with celestia-qt and with celestia-qt6.
With celestia-sdl the display is correct again in Plasma-Wayland.

If I start celestia-qt or celestia-qt6 with
`$ QT_QPA_PLATFORM=xcb celestia-qt`
the display is correct again, because then X11 is used for Celestia in the Wayland session.

For further details/testing, I need more instructions from the pros here on how to proceed.

In the meantime, celestia-qt or celestia-qt6 can be started and used as described above with the environment variable `QT_QPA_PLATFORM=xcb`.

The tests were done in a VMware environment, as I currently have a major problem running Celestia natively on my aged notebook: Shortly after startup, the entire computer freezes and only powering off/on brings it back to life.
I'll open a new issue about this when I have more info.

```
$ zypper se -si wayland qt celestia x11 
Loading repository data...
Reading installed packages...

S  | Name                                     | Type    | Version                         | Arch   | Repository
---+------------------------------------------+---------+---------------------------------+--------+-------------------------------------------
i  | attica-qt5                               | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | bluez-qt-imports                         | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | bluez-qt-udev                            | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | celestia                                 | package | 1.7.0~git20230224+15df4ba-708.1 | x86_64 | home:munix9:unstable (openSUSE_Tumbleweed)
i  | celestia-common                          | package | 1.7.0~git20230224+15df4ba-708.1 | noarch | home:munix9:unstable (openSUSE_Tumbleweed)
i  | celestia-data                            | package | 1.7.0~git20230201+ffc806d-86.1  | x86_64 | home:munix9:unstable (openSUSE_Tumbleweed)
i  | celestia-data-common                     | package | 1.7.0~git20230201+ffc806d-86.1  | noarch | home:munix9:unstable (openSUSE_Tumbleweed)
i  | celestia-data-lang                       | package | 1.7.0~git20230201+ffc806d-86.1  | noarch | home:munix9:unstable (openSUSE_Tumbleweed)
i  | celestia-lang                            | package | 1.7.0~git20230224+15df4ba-708.1 | noarch | home:munix9:unstable (openSUSE_Tumbleweed)
i+ | celestia-qt                              | package | 1.7.0~git20230224+15df4ba-708.1 | x86_64 | home:munix9:unstable (openSUSE_Tumbleweed)
i+ | celestia-qt6                             | package | 1.7.0~git20230224+15df4ba-708.1 | x86_64 | home:munix9:unstable (openSUSE_Tumbleweed)
i+ | celestia-sdl                             | package | 1.7.0~git20230224+15df4ba-708.1 | x86_64 | home:munix9:unstable (openSUSE_Tumbleweed)
i  | dbus-1-x11                               | package | 1.14.6-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | ghostscript-x11                          | package | 9.56.1-1.4                      | x86_64 | Haupt-Repository (OSS)
i  | kwayland                                 | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | kwayland-integration                     | package | 5.27.0-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | layer-shell-qt5                          | package | 5.27.0-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | libaccounts-qt5-1                        | package | 1.16-1.20                       | x86_64 | Haupt-Repository (OSS)
i  | libAppStreamQt2                          | package | 0.16.0-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | libcelestia1_7                           | package | 1.7.0~git20230224+15df4ba-708.1 | x86_64 | home:munix9:unstable (openSUSE_Tumbleweed)
i  | libdbusmenu-qt5-2                        | package | 0.9.3+16.04.20160218-1.12       | x86_64 | Haupt-Repository (OSS)
i  | libkdegames-qt5-imports                  | package | 22.12.2-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | libKF5BluezQt6                           | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | libKF5ModemManagerQt6                    | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | libKF5NetworkManagerQt6                  | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | libpackagekitqt5-1                       | package | 1.0.2-3.3                       | x86_64 | Haupt-Repository (OSS)
i  | libphonon4qt5                            | package | 4.11.1-3.7                      | x86_64 | Haupt-Repository (OSS)
i  | libpolkit-qt5-1-1                        | package | 0.114.0-2.8                     | x86_64 | Haupt-Repository (OSS)
i  | libpoppler-qt5-1                         | package | 23.02.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | libqaccessibilityclient-qt5-0            | package | 0.4.1-1.13                      | x86_64 | Haupt-Repository (OSS)
i  | libqca-qt5-2                             | package | 2.3.5-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qdbus                             | package | 5.15.8+kde1-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtbase-platformtheme-gtk3         | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtgraphicaleffects                | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtimageformats                    | package | 5.15.8+kde6-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtpaths                           | package | 5.15.8+kde1-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtquickcontrols                   | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtquickcontrols2                  | package | 5.15.8+kde7-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtstyleplugins-platformtheme-gtk2 | package | 5.0.0+git20170311-10.5          | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qttranslations                    | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtvirtualkeyboard                 | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtvirtualkeyboard-hunspell        | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtwayland                         | package | 5.15.8+kde57-1.1                | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtwebengine                       | package | 5.15.12-1.4                     | x86_64 | Haupt-Repository (OSS)
i  | libqt5-qtxmlpatterns-imports             | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Concurrent5                        | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5Core5                              | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5DBus5                              | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5Gui5                               | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5HunspellInputMethod5               | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libqt5keychain1                          | package | 0.13.2-3.5                      | x86_64 | Haupt-Repository (OSS)
i  | libqt5keychain1-lang                     | package | 0.13.2-3.5                      | noarch | Haupt-Repository (OSS)
i  | libQt5Multimedia5                        | package | 5.15.8+kde1-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Network5                           | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5NetworkAuth5                       | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5OpenGL5                            | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5Positioning5                       | package | 5.15.8+kde3-1.2                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5PrintSupport5                      | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5QuickControls2-5                   | package | 5.15.8+kde7-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5QuickTemplates2-5                  | package | 5.15.8+kde7-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Script5                            | package | 5.15.12-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | libQt5Sensors5                           | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Sensors5-imports                   | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Sql5                               | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5Sql5-mysql                         | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5Sql5-sqlite                        | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5Svg5                               | package | 5.15.8+kde8-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Test5                              | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5TextToSpeech5                      | package | 5.15.8+kde1-1.2                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5VirtualKeyboard5                   | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5WaylandClient5                     | package | 5.15.8+kde57-1.1                | x86_64 | Haupt-Repository (OSS)
i  | libQt5WaylandCompositor5                 | package | 5.15.8+kde57-1.1                | x86_64 | Haupt-Repository (OSS)
i  | libQt5WebChannel5                        | package | 5.15.8+kde3-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5WebChannel5-imports                | package | 5.15.8+kde3-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5WebView5                           | package | 5.15.8+kde0-1.2                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Widgets5                           | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5X11Extras5                         | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt5Xml5                               | package | 5.15.8+kde160-1.3               | x86_64 | Haupt-Repository (OSS)
i  | libQt5XmlPatterns5                       | package | 5.15.8+kde0-1.1                 | x86_64 | Haupt-Repository (OSS)
i  | libQt6Core5Compat6                       | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Core6                              | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6DBus6                              | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Gui6                               | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6LabsAnimation6                     | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6LabsFolderListModel6               | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6LabsQmlModels6                     | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6LabsSettings6                      | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6LabsSharedImage6                   | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6LabsWavefrontMesh6                 | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Network6                           | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6OpenGL6                            | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6OpenGLWidgets6                     | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Qml6                               | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QmlCore6                           | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QmlLocalStorage6                   | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QmlModels6                         | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QmlWorkerScript6                   | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QmlXmlListModel6                   | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Quick6                             | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickControls2-6                   | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickControls2Impl6                | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickDialogs2-6                    | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickDialogs2QuickImpl6            | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickDialogs2Utils6                | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickLayouts6                      | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickParticles6                    | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickShapes6                       | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickTemplates2-6                  | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6QuickTest6                         | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Sql6                               | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Test6                              | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6WaylandClient6                     | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6WaylandCompositor6                 | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6WaylandEglClientHwIntegration6     | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6WaylandEglCompositorHwIntegration6 | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6Widgets6                           | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQt6WlShellIntegration6                | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libQtQuick5                              | package | 5.15.8+kde22-1.1                | x86_64 | Haupt-Repository (OSS)
i  | libsignon-qt5-1                          | package | 8.60-2.18                       | x86_64 | Haupt-Repository (OSS)
i  | libva-wayland2                           | package | 2.17.0-1.2                      | x86_64 | Haupt-Repository (OSS)
i  | libva-x11-2                              | package | 2.17.0-1.2                      | x86_64 | Haupt-Repository (OSS)
i  | libwayland-client0                       | package | 1.21.0-1.2                      | x86_64 | Haupt-Repository (OSS)
i  | libwayland-cursor0                       | package | 1.21.0-1.2                      | x86_64 | Haupt-Repository (OSS)
i  | libwayland-egl1                          | package | 1.21.0-1.2                      | x86_64 | Haupt-Repository (OSS)
i  | libwayland-server0                       | package | 1.21.0-1.2                      | x86_64 | Haupt-Repository (OSS)
i  | libX11-6                                 | package | 1.8.1-4.1                       | x86_64 | Haupt-Repository (OSS)
i  | libX11-data                              | package | 1.8.1-4.1                       | noarch | Haupt-Repository (OSS)
i  | libX11-xcb1                              | package | 1.8.1-4.1                       | x86_64 | Haupt-Repository (OSS)
i  | libxkbcommon-x11-0                       | package | 1.5.0-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | libyui-qt-pkg16                          | package | 4.4.8-1.2                       | x86_64 | Haupt-Repository (OSS)
i  | libyui-qt16                              | package | 4.4.8-1.1                       | x86_64 | Haupt-Repository (OSS)
i+ | patterns-base-x11                        | package | 20200505-38.1                   | x86_64 | Haupt-Repository (OSS)
i  | patterns-base-x11_enhanced               | package | 20200505-38.1                   | x86_64 | Haupt-Repository (OSS)
i+ | patterns-yast-x11_yast                   | package | 20220411-1.3                    | x86_64 | Haupt-Repository (OSS)
i  | perl-X11-Protocol                        | package | 0.56-15.16                      | noarch | Haupt-Repository (OSS)
i  | phonon4qt5-backend-gstreamer             | package | 4.10.0-1.18                     | x86_64 | Haupt-Repository (OSS)
i  | phonon4qt5-backend-gstreamer-lang        | package | 4.10.0-1.18                     | noarch | Haupt-Repository (OSS)
i  | phonon4qt5-lang                          | package | 4.11.1-3.7                      | noarch | Haupt-Repository (OSS)
i  | pinentry-qt5                             | package | 1.2.1-1.3                       | x86_64 | Haupt-Repository (OSS)
i+ | plasma-wayland-protocols                 | package | 1.10-1.1                        | x86_64 | Haupt-Repository (OSS)
i  | plasma5-session-wayland                  | package | 5.27.0-2.1                      | x86_64 | Haupt-Repository (OSS)
i  | prison-qt5-imports                       | package | 5.103.0-1.1                     | x86_64 | Haupt-Repository (OSS)
i  | pulseaudio-module-x11                    | package | 16.1-2.1                        | x86_64 | Haupt-Repository (OSS)
i  | qca-qt5                                  | package | 2.3.5-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | qca-qt5-plugins                          | package | 2.3.5-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | qt6-declarative-imports                  | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | qt6-imageformats                         | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | qt6-network-tls                          | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | qt6-platformtheme-gtk3                   | package | 6.4.2-2.1                       | x86_64 | Haupt-Repository (OSS)
i  | qt6-translations                         | package | 6.4.2-1.1                       | noarch | Haupt-Repository (OSS)
i  | qt6-wayland                              | package | 6.4.2-1.1                       | x86_64 | Haupt-Repository (OSS)
i  | qtdeclarative-imports-provides-qt5       | package | 1.0-1.16                        | x86_64 | Haupt-Repository (OSS)
i  | vlc-qt                                   | package | 3.0.18-4.5                      | x86_64 | Haupt-Repository (OSS)
i  | wayland-utils                            | package | 1.1.0-1.1                       | x86_64 | Haupt-Repository (OSS)
i+ | x11                                      | pattern | 20200505-38.1                   | x86_64 | Haupt-Repository (OSS)
i  | x11-tools                                | package | 0.1-213.1                       | noarch | Haupt-Repository (OSS)
i+ | x11_yast                                 | pattern | 20220411-1.3                    | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-driver-video                    | package | 7.6_1-19.9                      | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-essentials                      | package | 7.6_1-16.17                     | noarch | Haupt-Repository (OSS)
i  | xorg-x11-fonts                           | package | 7.6-44.1                        | noarch | Haupt-Repository (OSS)
i  | xorg-x11-fonts-converted                 | package | 7.6-44.1                        | noarch | Haupt-Repository (OSS)
i  | xorg-x11-fonts-core                      | package | 7.6-44.1                        | noarch | Haupt-Repository (OSS)
i  | xorg-x11-fonts-legacy                    | package | 7.6-44.1                        | noarch | Haupt-Repository (OSS)
i  | xorg-x11-libX11-ccache                   | package | 7.6-22.4                        | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-server                          | package | 21.1.7-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-server-extra                    | package | 21.1.7-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-server-Xvfb                     | package | 21.1.7-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-Xvnc                            | package | 1.13.0-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | xorg-x11-Xvnc-module                     | package | 1.13.0-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | xwayland                                 | package | 22.1.8-1.1                      | x86_64 | Haupt-Repository (OSS)
i  | yast2-control-center-qt                  | package | 4.5.0-1.4                       | x86_64 | Haupt-Repository (OSS)
i  | yast2-qt-branding-openSUSE               | package | 84.87.20210910-1.37             | noarch | Haupt-Repository (OSS)
i  | yast2-x11                                | package | 4.5.1-1.2                       | x86_64 | Haupt-Repository (OSS)
```

**Comment by munix9** [comment]
Another test, but the result does not necessarily help to find a solution:

I install `weston`, the Wayland Reference Compositor, start it in a Plasma-X11 session and then inside weston a console is started and here again I start `celestia-qt`.

And here the transparency works normally again.
I can't make any sense of it right now.

```
$ sudo zypper in weston
The following 7 NEW packages are going to be installed:
  libfreerdp2-2 libseat1 libweston-11 libweston-11-0 libwinpr2-2 vpx-tools weston
$ weston
```
Inside weston from the top left start a terminal and then
`$ celestia-qt`

**Comment by levinli303** [comment]
please post the render info panel compiled with the latest commit.

**Comment by munix9** [comment]
```
OpenGL version: 3.3 (Compatibility Profile) Mesa 22.3.5
Vendor: VMware, Inc.
Renderer: SVGA3D; build: RELEASE; LLVM;
GLSL Version: 3.30
Max simultaneous textures: 48
Maximum texture size: 16384
Point size range: 1 - 80
Point size granularity: 0.10
Number of interpolators: 124
Max anisotropy filtering: 16.00

Supported extensions:
GL_ARB_multisample
GL_EXT_abgr
GL_EXT_bgra
GL_EXT_blend_color
GL_EXT_blend_minmax
GL_EXT_blend_subtract
GL_EXT_copy_texture
GL_EXT_subtexture
GL_EXT_texture_object
GL_EXT_vertex_array
GL_EXT_compiled_vertex_array
GL_EXT_texture
GL_EXT_texture3D
GL_IBM_rasterpos_clip
GL_ARB_point_parameters
GL_EXT_draw_range_elements
GL_EXT_packed_pixels
GL_EXT_point_parameters
GL_EXT_rescale_normal
GL_EXT_separate_specular_color
GL_EXT_texture_edge_clamp
GL_SGIS_generate_mipmap
GL_SGIS_texture_border_clamp
GL_SGIS_texture_edge_clamp
GL_SGIS_texture_lod
GL_ARB_framebuffer_sRGB
GL_ARB_multitexture
GL_EXT_framebuffer_sRGB
GL_IBM_multimode_draw_arrays
GL_IBM_texture_mirrored_repeat
GL_ARB_texture_cube_map
GL_ARB_texture_env_add
GL_ARB_transpose_matrix
GL_EXT_blend_func_separate
GL_EXT_fog_coord
GL_EXT_multi_draw_arrays
GL_EXT_secondary_color
GL_EXT_texture_env_add
GL_EXT_texture_filter_anisotropic
GL_EXT_texture_lod_bias
GL_INGR_blend_func_separate
GL_NV_blend_square
GL_NV_light_max_exponent
GL_NV_texgen_reflection
GL_NV_texture_env_combine4
GL_S3_s3tc
GL_SUN_multi_draw_arrays
GL_ARB_texture_border_clamp
GL_ARB_texture_compression
GL_EXT_framebuffer_object
GL_EXT_texture_compression_s3tc
GL_EXT_texture_env_combine
GL_EXT_texture_env_dot3
GL_MESA_window_pos
GL_NV_packed_depth_stencil
GL_NV_texture_rectangle
GL_ARB_depth_texture
GL_ARB_occlusion_query
GL_ARB_shadow
GL_ARB_texture_env_combine
GL_ARB_texture_env_crossbar
GL_ARB_texture_env_dot3
GL_ARB_texture_mirrored_repeat
GL_ARB_window_pos
GL_ATI_fragment_shader
GL_EXT_stencil_two_side
GL_EXT_texture_cube_map
GL_NV_depth_clamp
GL_NV_fog_distance
GL_NV_half_float
GL_APPLE_packed_pixels
GL_ARB_draw_buffers
GL_ARB_fragment_program
GL_ARB_fragment_shader
GL_ARB_shader_objects
GL_ARB_vertex_program
GL_ARB_vertex_shader
GL_ATI_draw_buffers
GL_ATI_texture_env_combine3
GL_ATI_texture_float
GL_EXT_shadow_funcs
GL_EXT_stencil_wrap
GL_MESA_pack_invert
GL_NV_primitive_restart
GL_ARB_depth_clamp
GL_ARB_fragment_program_shadow
GL_ARB_half_float_pixel
GL_ARB_occlusion_query2
GL_ARB_point_sprite
GL_ARB_shading_language_100
GL_ARB_sync
GL_ARB_texture_non_power_of_two
GL_ARB_vertex_buffer_object
GL_ATI_blend_equation_separate
GL_EXT_blend_equation_separate
GL_OES_read_format
GL_ARB_color_buffer_float
GL_ARB_pixel_buffer_object
GL_ARB_texture_compression_rgtc
GL_ARB_texture_float
GL_ARB_texture_rectangle
GL_EXT_packed_float
GL_EXT_pixel_buffer_object
GL_EXT_texture_compression_dxt1
GL_EXT_texture_compression_rgtc
GL_EXT_texture_rectangle
GL_EXT_texture_sRGB
GL_EXT_texture_shared_exponent
GL_ARB_framebuffer_object
GL_EXT_framebuffer_blit
GL_EXT_framebuffer_multisample
GL_EXT_packed_depth_stencil
GL_ARB_vertex_array_object
GL_ATI_separate_stencil
GL_EXT_draw_buffers2
GL_EXT_draw_instanced
GL_EXT_gpu_program_parameters
GL_EXT_gpu_shader4
GL_EXT_texture_array
GL_EXT_texture_integer
GL_EXT_texture_sRGB_decode
GL_EXT_timer_query
GL_OES_EGL_image
GL_EXT_texture_buffer_object
GL_ARB_copy_buffer
GL_ARB_depth_buffer_float
GL_ARB_draw_instanced
GL_ARB_half_float_vertex
GL_ARB_instanced_arrays
GL_ARB_map_buffer_range
GL_ARB_texture_buffer_object
GL_ARB_texture_rg
GL_ARB_texture_swizzle
GL_ARB_vertex_array_bgra
GL_EXT_texture_swizzle
GL_EXT_vertex_array_bgra
GL_NV_conditional_render
GL_AMD_conservative_depth
GL_ARB_ES2_compatibility
GL_ARB_blend_func_extended
GL_ARB_compatibility
GL_ARB_debug_output
GL_ARB_draw_elements_base_vertex
GL_ARB_explicit_attrib_location
GL_ARB_fragment_coord_conventions
GL_ARB_provoking_vertex
GL_ARB_sampler_objects
GL_ARB_seamless_cube_map
GL_ARB_shader_texture_lod
GL_ARB_texture_buffer_object_rgb32
GL_ARB_texture_multisample
GL_ARB_texture_rgb10_a2ui
GL_ARB_uniform_buffer_object
GL_ARB_vertex_type_2_10_10_10_rev
GL_EXT_provoking_vertex
GL_EXT_texture_snorm
GL_MESA_texture_signed_rgba
GL_NV_copy_image
GL_ARB_get_program_binary
GL_ARB_robustness
GL_ARB_separate_shader_objects
GL_ARB_shader_bit_encoding
GL_ARB_shader_subroutine
GL_ARB_timer_query
GL_ARB_viewport_array
GL_EXT_direct_state_access
GL_ANGLE_texture_compression_dxt3
GL_ANGLE_texture_compression_dxt5
GL_ARB_compressed_texture_pixel_storage
GL_ARB_conservative_depth
GL_ARB_internalformat_query
GL_ARB_map_buffer_alignment
GL_ARB_shading_language_420pack
GL_ARB_shading_language_packing
GL_ARB_texture_storage
GL_EXT_framebuffer_multisample_blit_scaled
GL_EXT_transform_feedback
GL_AMD_shader_trinary_minmax
GL_ARB_ES3_compatibility
GL_ARB_arrays_of_arrays
GL_ARB_clear_buffer_object
GL_ARB_copy_image
GL_ARB_explicit_uniform_location
GL_ARB_fragment_layer_viewport
GL_ARB_invalidate_subdata
GL_ARB_program_interface_query
GL_ARB_stencil_texturing
GL_ARB_texture_buffer_range
GL_ARB_texture_query_levels
GL_ARB_texture_storage_multisample
GL_ARB_vertex_attrib_binding
GL_KHR_debug
GL_KHR_texture_compression_astc_ldr
GL_ARB_buffer_storage
GL_ARB_clear_texture
GL_ARB_internalformat_query2
GL_ARB_multi_bind
GL_ARB_shading_language_include
GL_ARB_texture_stencil8
GL_EXT_debug_label
GL_EXT_shader_integer_mix
GL_ARB_direct_state_access
GL_ARB_get_texture_sub_image
GL_KHR_context_flush_control
GL_ARB_parallel_shader_compile
GL_KHR_no_error
GL_KHR_texture_compression_astc_sliced_3d
GL_MESA_shader_integer_functions
GL_ARB_texture_filter_anisotropic
GL_KHR_parallel_shader_compile
GL_EXT_EGL_image_storage
GL_EXT_EGL_sync
GL_NV_ES1_1_compatibility
```

**Comment by levinli303** [comment]
need to build with commit 15df4ba

**Comment by munix9** [comment]
hm, strange, this is 15df4ba, see https://github.com/CelestiaProject/Celestia/issues/1601#issuecomment-1445047569

**Comment by levinli303** [comment]
hmm there should be a line with `Color component` if it is compiled against https://github.com/CelestiaProject/Celestia/commit/15df4ba74b3ae5364cc470c262da4cbd6c1ebc9c

**Comment by munix9** [comment]
Maybe I'm in the wrong place: how exactly do I get the output I want?

**Comment by munix9** [comment]
or is the output miising in `src/celestia/qt/qtappwin.cpp` ?

**Comment by levinli303** [comment]
yeah it is missing there.

**Comment by munix9** [comment]
In the meantime, I'll test a local build and revert https://github.com/CelestiaProject/Celestia/pull/1594, maybe it's because of that, as 375gnu noted above.
But this will take some time.

**Comment by levinli303** [comment]
try adding these to line 1186

```
    // shading language version
    if (info.count("ColorComponent") > 0)
    {
        out << QString(_("<b>Color component:</b> %1")).arg(info["ColorComponent"].c_str());
        out << "<br>\n";
    }
```

**Comment by ajtribick** [comment]
Looking again at #1594 - I *think* an alternate fix might be to change https://github.com/CelestiaProject/Celestia/blob/15df4ba74b3ae5364cc470c262da4cbd6c1ebc9c/src/celengine/render.cpp#L5410

to use `glBlendFuncSeparate(ps.blendFunc.src, ps.blendFunc.dst, GL_ONE, GL_ONE_MINUS_SRC_ALPHA)`

this at least stops the Milky Way from changing the framebuffer's alpha. Whether this works for all our transparent rendering is another matter

**Comment by munix9** [comment]
Now with "Color component".
The only difference between the outputs is

`x11:     Color component: RGB888`
`wayland: Color component: RGBA8888`

Reverting https://github.com/CelestiaProject/Celestia/pull/1594 did not help.

**Comment by levinli303** [comment]
> Now with "Color component". The only difference between the outputs is
> 
> `x11: Color component: RGB888` `wayland: Color component: RGBA8888`

Wayland has this bug right? Then it is the same as #1551 which we need to fix.

**Comment by munix9** [comment]
Wayland, yes, more precisely: Wayland in the current openSUSE Tumbleweed version.
According to Askaniy, however, the representation in the AppImage is correct, and this was created with openSUSE Leap 15.4.

I will also run tests in a VM with openSUSE Leap 15.4 (Xorg/Wayland), but I suspect that more recent packages in Tumbleweed are causing the display problems.

**Comment by munix9** [comment]
As I suspected: in openSUSE Leap 15.4, Wayland Plasma session, the transparency display is correct.

However, the mouse does not work correctly, I can not rotate the earth - but that is probably another problem.

**Comment by 375gnu** [comment]
> `x11: Color component: RGB888` `wayland: Color component: RGBA8888`
> 
> Reverting #1594 did not help.

X11 behaves correctly with #1594 but wayland ignores request to avoid alpha. This is either wayland or qt bug.

**Comment by Askaniy** [comment]
It seems better after #1605
![изображение](https://user-images.githubusercontent.com/44873271/221519018-43fb01db-5989-4e4a-b9fc-655f41ae694e.png)

**Comment by Askaniy** [comment]
But now we have aliased edge of the atmosphere and some fade effect (slow motion video)

https://user-images.githubusercontent.com/44873271/221528714-46d5894e-77c4-4434-85bd-f98f015a1226.mp4

There is a chance that this is just an artifact of my VA monitor, but I have not seen this before

**Comment by Askaniy** [comment]
The atmosphere is just still transparent:
![изображение](https://user-images.githubusercontent.com/44873271/221542511-74dbf65d-be8e-49e8-b300-5b3c542498e2.png)
@ajtribick, is it possible to repeat #1605 for atmospheres?

**Comment by munix9** [comment]
I still think it's due to current packages (Qt, Wayland) in openSUSE Tumbleweed, as the rendering in Leap 15.4 is correct (AppImage).

Either a bug has crept into the very current packages of Qt/Wayland, or something new in the versions leads to the wrong display.

It is also possible that the problem will be solved in one of the next Tumbleweed versions, but I don't know.

**Comment by Askaniy** [comment]
This problem sometimes shows up on other platforms, so I think if we are able (?) to fix it now, it may prevent similar bugs in the future.

**Comment by levinli303** [comment]
The root cause is known however it might be just hard to fix.

On the frontends i maintain, there's workaround. iOS/macOS native (set layer to opacity), Android/UWP (find an opaque OpenGL config). I don't know if there is similar workaround on QT.

**Comment by ajtribick** [comment]
Good to see the continents are working for you. I suspect the change I mentioned earlier in this thread to use `glBlendFuncSeparate` to avoid writing the alpha channel is probably still necessary then, since this also worked for me in solving the transparent image capture issue on Windows. Hopefully will have time to look at this in the next few days.

**Comment by 375gnu** [comment]
@Askaniy please open a terminal and run the following command:
```
QT_LOGGING_RULES=qt.qpa.gl=true celestia-qt
```
And then post output here.

**Comment by ajtribick** [comment]
Is this still occurring since merging #1616?

**Comment by munix9** [comment]
On my openSUSE Tumbleweed 20230302 it looks good now (celestia-1.7.0~git20230303+f70fc01-718.1.x86_64), at least what I can see until the computer completely LOCKs.

I had hinted before that my old notebook has been freezing completely for some time when I start Celestia natively. If I start the AppImage it runs.

Let's see if I can debug something there, unfortunately nothing is written in the logs at all (Plasma, Xorg, i915, Intel HD Graphics 4000, Kernel 6.2.1).

**Comment by Askaniy** [comment]
#1616 works well on the latest openSUSE Tumbleweed build 👍
![изображение](https://user-images.githubusercontent.com/44873271/222906024-ba3efa85-faa7-4b7d-b63c-9051c5826c32.png)

**Comment by Askaniy** [comment]
Apparently, the topic issue is done, but
- Should workarounds be removed on other platforms? Suddenly it will increase performance, but it's just thinking out loud
- No output logs on my Linux configuration and on Windows 11 too, and at least 2 more people with no logs on Linux. Probably it should be a new issue
